addons/google_calendar/tests/test_sync_google2odoo.py

```diff
@@ -980,6 +980,195 @@ def test_simple_event_into_recurrency(self):
+                            "email": self.organizer_user.partner_id.email,
+                            "responseStatus": "accepted",
+                        },
+                        {
+                            "email": self.attendee_user.partner_id.email,
+                            "responseStatus": "accepted",
+                        },
+                    ],
+                },
+                # Event details for day 14
+                {
+                    "id": "%s_20240614T080000Z" % google_id,
+                    "summary": "coucou",
+                    "start": {"dateTime": "2024-06-14T08:00:00+00:00"},
+                    "end": {"dateTime": "2024-06-14T10:00:00+00:00"},
+                    "reminders": {"useDefault": True},
+                    "updated": self.now,
+                    "recurringEventId": "%s_R20240614T080000" % google_id,
+                    "attendees": [
+                        {
+                            "email": self.organizer_user.partner_id.email,
+                            "responseStatus": "accepted",
+                        },
+                        {
+                            "email": self.attendee_user.partner_id.email,
+                            "responseStatus": "accepted",
+                        },
+                    ],
+                },
+                # New recurrence that starts on day 14
+                {
+                    "id": "%s_R20240614T080000" % google_id,
+                    "summary": "coucou",
+                    "start": {"dateTime": "2024-06-14T08:00:00+00:00"},
+                    "end": {"dateTime": "2024-06-14T10:00:00+00:00"},
+                    "recurrence": [
+                        "RRULE:FREQ=WEEKLY;WKST=MO;UNTIL=20240620T215959Z;BYDAY=FR"
+                    ],
+                    "reminders": {"useDefault": True},
+                    "updated": self.now,
+                    "attendees": [
+                        {
+                            "email": self.organizer_user.partner_id.email,
+                            "responseStatus": "accepted",
+                        },
+                        {
+                            "email": self.attendee_user.partner_id.email,
+                            "responseStatus": "accepted",
+                        },
+                    ],
+                },
+            ]
+            # Then, Odoo syncs
+            with patch.object(
+                GoogleCalendarService,
+                "get_events",
+                return_value=(
+                    GoogleEvent(values),
+                    None,
+                    [{"method": "popup", "minutes": 30}],
+                ),
+            ):
+                self.attendee_user.sudo()._sync_google_calendar(self.google_service)
+            events = self.env["calendar.event"].search(
+                [("google_id", "like", google_id)]
+            )
+            self.assertEqual(len(events.exists()), 2)
+
```

Comment:
Why are you using those weird patches ?
I think it is easier to do it that way:
value_list = [...]
self.env['calendar.event']._sync_google2odoo(GoogleEvent(value_list))

You know that you will sync from google to odoo. No need to call _sync_google2odoo then _sync_odoo2google in this _sync_google_calendar call. It is more clear , lighter and faster.